packages/repository/src/relations/relation.helpers.ts

```diff
@@ -48,3 +58,85 @@ export async function findByForeignKeys<
 }
 
 type StringKeyOf<T> = Extract<keyof T, string>;
+
+/**
+ * Returns model instances that include related models that have a registered
+ * resolver.
+ *
+ * @param targetRepository - The target repository where the model instances are found
+ * @param entities - An array of entity instances or data
+ * @param filter - Query filter
+ * @param options - Options for the operations
+ */
+
+export async function includeRelatedModels<
+  T extends Entity,
+  Relations extends object = {}
+>(
+  targetRepository: EntityCrudRepository<T, unknown, Relations>,
+  entities: T[],
+  filter?: Filter<T>,
+  options?: Options,
+): Promise<(T & Relations)[]> {
+  const result = entities as (T & Relations)[];
+  const include = filter && filter.include;
+  if (!include) return result;
+
+  const invalidInclusions = include.filter(
+    inclusionFilter => !isInclusionAllowed(targetRepository, inclusionFilter),
+  );
+  if (invalidInclusions.length) {
+    const msg =
+      'Invalid "filter.include" entries: ' +
+      invalidInclusions
+        .map(inclusionFilter => JSON.stringify(inclusionFilter))
+        .join('; ');
+    const err = new Error(msg);
+    Object.assign(err, {
+      code: 'INVALID_INCLUSION_FILTER',
+    });
+    throw err;
+  }
+
+  const resolveTasks = include.map(async inclusionFilter => {
+    const relationName = inclusionFilter.relation;
+    const resolver = targetRepository.inclusionResolvers.get(relationName)!;
+    const targets = await resolver(entities, inclusionFilter, options);
+
+    for (const ix in result) {
```

Comment:
Don't use `for in` loop for an array!

Reply:
Can we do:
```ts
result.forEach((entity, ix) => {
  const src = entity as AnyObject;
  src[relationName] = targets[ix];
});
```
? We need the index to get the element in `targets` that's for that specific source entity.